### DISCUSSION TITLE
How to excavate on the established model?

### DISCUSSION THREAD

**Discussion by zuoyu158** [gempy-project/gempy#653] [Q&A] [answered]
How to excavate on the established model?
How to build a tunnel?
Is there any direction for mining engineering? May I ask you some questions?

**Comment by AlexanderJuestel** [comment, marked as answer]
Hello @zuoyu158, 

GemPy currently does not support any mining engineering tasks as it is made for structural geological modelling in the first place. 

As the 3D visualization is in Pyvista, you could add a tunnel yourself by creating a tube that represents the tunnel. Let me know if you need any instructions for that. 

Cheers Alex

**Reply by zuoyu158** [reply]
Hi
Thanks for your answer. How to create a tube that represents a tunnel？

**Reply by AlexanderJuestel** [reply]
@zuoyu158 
Have a look at this little tutorial here using PyVista: https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/53_adding_anthropogenic_geometries_to_pyvista.html

**Reply by zuoyu158** [reply]
Hi@AlexanderJuestel
Thank you very much. I can not find the data of 53_adding_anthropogenic_geometries_to_pyvista. Can you help me？

**Reply by AlexanderJuestel** [reply]
Hello @zuoyu158,

The way of defining this tunnel object is shown starting from here using a NumPy array and PyVista: https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/53_adding_anthropogenic_geometries_to_pyvista.html#Defining-Tunnel-Object 

You can just apply it to your data ;)

**Reply by zuoyu158** [reply]
Thank you.
